platform/jobs/edxEndToEndTestsPullRequestJob.groovy

```diff
@@ -0,0 +1,183 @@
+    String ghprbConfigContents = new File("${GHPRB_SECRET}").text
+    Yaml yaml = new Yaml()
+    secretMap = yaml.load(secretFileContents)
+    ghprbMap = yaml.load(ghprbConfigContents)
+    out.println('Successfully parsed secret YAML file')
+}
+catch (any) {
+    out.println('Jenkins DSL: Error parsing secret YAML file')
+    out.println('Exiting with error code 1')
+    return 1
+}
+
+/* Iterate over the job configurations */
+secretMap.each { jobConfigs ->
+
+    Map jobConfig = jobConfigs.getValue()
+
+    /* Test secrets contain all necessary keys for this job */
+    /* TODO: Use/Build a more robust test framework for this */
+    assert jobConfig.containsKey('authUser')
+    assert jobConfig.containsKey('authPass')
+    assert jobConfig.containsKey('loginEmail')
+    assert jobConfig.containsKey('loginPass')
+    assert jobConfig.containsKey('open')
+    assert jobConfig.containsKey('email')
+    assert jobConfig.containsKey('githubRepoUrlSegment')
+    assert jobConfig.containsKey('jobName')
+    assert jobConfig.containsKey('platformCredential')
+
+    assert ghprbMap.containsKey('admin')
+    assert ghprbMap.containsKey('userWhiteList')
+    assert ghprbMap.containsKey('orgWhiteList')
+
+    job(jobConfig['jobName']) {
+
+        /* For non-open jobs, enable project based security */
+        if (!jobConfig['open'].toBoolean()) {
+            authorization {
+                blocksInheritance(true)
+                permissionAll('edx')
+            }
+        }
+
+        parameters {
+            stringParams.each { param ->
+                stringParam(param.name, param.default, param.description)
+            }
+        }
+
+        properties {
+              githubProjectUrl(JENKINS_PUBLIC_GITHUB_BASEURL + jobConfig['githubRepoUrlSegment'])
+        }
+
+        logRotator JENKINS_PUBLIC_LOG_ROTATOR() //Discard build after 14 days
+
+        label('jenkins-worker')
+
+        checkoutRetryCount(5)
+
+        environmentVariables {
+            env('BASIC_AUTH_USER', jobConfig['authUser'])
+            env('BASIC_AUTH_PASSWORD', jobConfig['authPass'])
+            env('USER_LOGIN_EMAIL', jobConfig['loginEmail'])
+            env('USER_LOGIN_PASSWORD', jobConfig['loginPass'])
+        }
+
+        scm {
+
+           git { //using git on the branch and url, clone, clean before checkout
+                remote {
+                    url(JENKINS_PUBLIC_GITHUB_BASEURL + jobConfig['githubRepoUrlSegment'] + '.git')
+                    refspec('+refs/pull/*:refs/remotes/origin/pr/*')
+                    if (!jobConfig['open'].toBoolean()) {
+                        credentials(jobConfig['platformCredential'])
+                    }
+                }
+                branch('${ghprbActualCommit}')
+                browser()
+            }
+        }
+
+        steps {
+            shell("jenkins/end_to_end_tests.sh")
+        }
+
+        triggers { //triggers on each git pull request
+          pullRequest {
```

Comment:
I think we should use useGitHubHooks() here. It will use web hooks instead of constantly polling github. For an example see  jenkins-job-dsl/platform/jobs/edxPlatformBokChoyPr.groovy.

We are using v1.45 of the DSL plugin (Sorry that that is not clear in the repo, we will change this). The Github Pull Request Builder is deprecated in more recent versions of the plugin, and therefore the API viewer (https://jenkinsci.github.io/job-dsl-plugin/#) must be set to the appropriate version (1.45) when looking up how to configure this in the DSL.